marketplaceItems/questionApp/appResources/appData/question_app.js

```diff
@@ -0,0 +1,238 @@
+            type: "Model",
+            modelURL: Script.resolvePath("resources/models/sphere-white-emissive.fbx"),
+            lifetime: 360000,
+            parentID: MyAvatar.sessionUUID,
+            parentJointIndex: parentJointIndex,
+            localPosition: { x: 0, y: questionMarkLocalYPosition, z: 0 },
+            dimensions: { x: 0.2, y: 0.2, z: 0.2 },
+            grab: { grabbable: false },
+            collisionless: true
+        }, 'avatar');
+        questionMarkMaterial = Entities.addEntity({
+            type: "Material",
+            name: "Question App Material",
+            materialURL: "materialData",
+            priority: 1,
+            parentID: questionMark,
+            materialData: JSON.stringify({
+                materials: {
+                    albedo: { red: 0, green: 255, blue: 0 },
+                    emissive: { red: 0, green: 1, blue: 0 }
+                }
+            })
+        }, 'avatar');
+        print("start growing");
+        growingInterval = Script.setInterval(function() {
+            var questionMarkDimensions = Entities.getEntityProperties(questionMark, 'dimensions').dimensions;
+            questionMarkDimensions = Vec3.multiply(questionMarkDimensions, GROWTH_RATIO);
+            Entities.editEntity(questionMark, { dimensions: questionMarkDimensions });
+            if (questionMarkDimensions.y > MAX_HEIGHT_M) {
+                print("stop growing");
+                Script.clearInterval(growingInterval);
+            }
+        }, GROWTH_INTERVAL_MS);
+    }
+
+    /* ON CLICKING APP BUTTON: (on the toolbar or tablet) if we are opening the app, play a sound and get the question mark.
+    If we are closing the app, remove the question mark and play a different sound */
+    var OPEN_SOUND = SoundCache.getSound(Script.resolvePath('resources/sounds/open.mp3'));
+    var OPEN_SOUND_VOLUME = 0.2;
+    var CLOSE_SOUND = SoundCache.getSound(Script.resolvePath('resources/sounds/close.mp3?0'));
+    var CLOSE_SOUND_VOLUME = 0.3;
+    var QUESTION_CHANNEL = "TriviaChannel";
+    function onClicked() {
+        if (questionMark) {
+            cleanUp();
+            playSound(CLOSE_SOUND, CLOSE_SOUND_VOLUME, MyAvatar.position, true, false);
+            Messages.messageReceived.disconnect(checkMessage);
+        } else {
+            button.editProperties({ isActive: true });
+            playSound(OPEN_SOUND, OPEN_SOUND_VOLUME, MyAvatar.position, true, false);
+            createQuestionMark();
+            Messages.messageReceived.connect(checkMessage);\
+        }
+    }
+
+    /* Handles avatar being selected */
+    pickRayController
+        .registerEventHandler(selectAvatar)
+        .setType("avatar")
+        .setMapName("hifi_question")
+        .create();
+
+    /* SELECT AVATAR: When an admin clicks/triggers on an avatar, they are selected to speak next and their UUID 
+    is sent out via message */
+    function selectAvatar(uuid, intersection) {
```

Comment:
it looks like this logic doesn't gate on `adminStatus` - why is that? it looks like, as this code is written, any avatar could make another user's question mark disappear.

Reply:
only an admin has the controller function that allows clicking on other avatars.